### PR TITLE
[Moved to openshift/console]integrate routing advanced option in git import flow

### PR DESCRIPTION
**Pull request by divyanshiGupta**
Story: https://jira.coreos.com/browse/ODC-1020

routes

### CODE REVIEW

**Comment by divyanshiGupta** [comment]
@christianvogt I know you asked us not to raise any PRs, but I just got finished with the work and thought that it will be a good idea to get the PR reviewed. I will raise a new PR on openshift console repo once we are merged.

File: frontend/public/extend/devconsole/components/formik-fields/field-types.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export interface DropdownFieldProps extends InputFieldProps {
   selectedKey: string;
   title?: React.ReactNode;
   fullWidth?: boolean;
+  emptyStateMsg?: string;
```
Comment (rohitkrai03) [inline frontend/public/extend/devconsole/components/formik-fields/field-types.ts]: Why do we need this?

File: frontend/public/extend/devconsole/components/import/__tests__/import-validation-utils.spec.ts [hunk 2]
```diff
@@ -19,79 +20,59 @@ describe('ValidationUtils', () => {
...
-        },
-        replicas: 1,
-      },
-      labels: {},
-    };
-
-    it('should validate the form data', async () => {
-      await validationSchema.isValid(mockFormData).then((valid) => expect(valid).toEqual(true));
+    it('should validate the form data', async() => {
+      const mockData = cloneDeep(mockFormData);
```
Comment (rohitkrai03) [inline frontend/public/extend/devconsole/components/import/__tests__/import-validation-utils.spec.ts:24]: Thanks for doing this. I was meaning to do it but missed it in the last PR.

File: frontend/public/extend/devconsole/components/import/GitImportForm.tsx [hunk 3]
```diff
@@ -34,7 +34,7 @@ const GitImportForm: React.FC<FormikProps<FormikValues> & GitImportFormProps> =
       <AppSection project={values.project} />
       <BuilderSection image={values.image} builderImages={builderImages} />
       <ExpandCollapse>
-        <RouteSection />
+        <RouteSection create={values.route.create} />
```
Comment (rohitkrai03) [inline frontend/public/extend/devconsole/components/import/GitImportForm.tsx]: Better to send `route` as prop here. Create doesn't give full context.

File: frontend/public/extend/devconsole/components/import/route/CreateRouteForm.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
+/* eslint-disable no-unused-vars, no-undef */
```
Comment (rohitkrai03) [inline frontend/public/extend/devconsole/components/import/route/CreateRouteForm.tsx]: Maybe just rename it to `CreateRoute.tsx`

File: frontend/public/extend/devconsole/components/import/route/CreateRouteForm.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          type="text"
+          name="route.path"
+          label="Path"
+          helpText="Path that the router watches to route traffic to the service."
+        />
+        <DropdownField
+          name="route.targetPort"
+          label="Target Port"
+          items={portOptions}
+          selectedKey=""
```
Comment (rohitkrai03) [inline frontend/public/extend/devconsole/components/import/route/CreateRouteForm.tsx]:
```suggestion
          selectedKey={targetPort}
```

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/public/extend/devconsole/components/formik-fields/field-types.ts (hunk 1 above)
Comment (divyanshiGupta) [reply frontend/public/extend/devconsole/components/formik-fields/field-types.ts]: Forgot to remove this. :P

File: frontend/public/extend/devconsole/components/import/__tests__/import-validation-utils.spec.ts (hunk 2 above)
Comment (divyanshiGupta) [reply frontend/public/extend/devconsole/components/import/__tests__/import-validation-utils.spec.ts:24]: You are welcome :)

File: frontend/public/extend/devconsole/components/import/route/CreateRouteForm.tsx (hunk 5 above)
Comment (divyanshiGupta) [reply frontend/public/extend/devconsole/components/import/route/CreateRouteForm.tsx]: Missed it. Thanks

**Comment by rohitkrai03** [comment]
@divyanshiGupta Can you also add `validateForm(newValues)` to `GitSection.handleGitUrlBlur()` and `BuilderImageSelecter.handleImageChange()`? I see some flickers of validation error messages sometimes. After downgrading to rc-1 i thought we won't need that part but we need it after all.

**Comment by divyanshiGupta** [comment]
> @divyanshiGupta Can you also add `validateForm(newValues)` to `GitSection.handleGitUrlBlur()` and `BuilderImageSelecter.handleImageChange()`? I see some flickers of validation error messages sometimes. After downgrading to rc-1 i thought we won't need that part but we need it after all.

Done @rohitkrai03

**Review by rohitkrai03 — APPROVED** [review]
LGTM

**Comment by sbose78** [comment]
build and deploy

**Comment by sbose78** [comment]
Your build is progress : https://console-openshift-console.apps.rohit37.devcluster.openshift.com/k8s/ns/openshift-console/builds
 
 PR will be LIVE in https://pr-141-openshift-console.apps.rohit37.devcluster.openshift.com

File: frontend/public/extend/devconsole/components/import/route/CreateRoute.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    acc[name] = (
+      <React.Fragment>
+        {port.containerPort} &rarr; {port.containerPort} ({port.protocol})
+      </React.Fragment>
+    );
+    return acc;
+  }, {});
+
+  return (
+    <React.Fragment>
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/import/route/CreateRoute.tsx]: Remove `React.Fragment`

File: frontend/public/extend/devconsole/components/import/route/SecureRoute.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  return (
+    <FormGroup>
+      <ControlLabel>Security</ControlLabel>
+      <CheckboxField
+        name="route.secure"
+        label="Secure Route"
+        helpText="Routes can be secured using several TLS termination types for serving certificates."
+      />
+      {secure && (
+        <div className="co-create-route__security">
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/import/route/SecureRoute.tsx]: Why is this here? What styles does it add as I cannot find any reference.
If not needed, please remove.

File: frontend/public/extend/devconsole/components/import/import-validation-utils.ts [hunk 8]
```diff
@@ -37,6 +39,26 @@ export const validationSchema = yup.object().shape({
...
+      }),
+    }),
+    hostname: yup
+      .string()
+      .matches(hostnameRegex, {
+        message:
+          'Hostname must consist of lower-case letters, numbers, periods, and hyphens. It must start and end with a letter or number.',
+        excludeEmptyString: true,
+      })
+      .max(253, 'Can\'t be longer than 253 characters.'),
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/import/import-validation-utils.ts]: For consistency with required message.
```suggestion
      .max(253, 'Cannot be longer than 253 characters.'),
```

File: frontend/public/extend/devconsole/components/import/import-validation-utils.ts [hunk 9]
```diff
@@ -37,6 +39,26 @@ export const validationSchema = yup.object().shape({
...
+    path: yup
+      .string()
+      .matches(pathRegex, { message: 'Path must start with /', excludeEmptyString: true }),
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/import/import-validation-utils.ts]: Could you make every message consistent and add a period to the end.

File: frontend/public/extend/devconsole/components/import/route/SecureRoute.tsx (hunk 7 above)
Comment (divyanshiGupta) [reply frontend/public/extend/devconsole/components/import/route/SecureRoute.tsx]: This was being used in console `create-route` but we don't need it anymore. Forgot to remove this.

File: frontend/public/extend/devconsole/components/import/import-validation-utils.ts (hunk 9 above)
Comment (divyanshiGupta) [reply frontend/public/extend/devconsole/components/import/import-validation-utils.ts]: Done

File: frontend/public/extend/devconsole/components/import/import-validation-utils.ts (hunk 8 above)
Comment (divyanshiGupta) [reply frontend/public/extend/devconsole/components/import/import-validation-utils.ts]: Done

File: frontend/public/extend/devconsole/components/import/route/CreateRoute.tsx (hunk 6 above)
Comment (divyanshiGupta) [reply frontend/public/extend/devconsole/components/import/route/CreateRoute.tsx]: Done

**Pull request closed without merging**